Library/Homebrew/os/mac/sdk.rb

```diff
@@ -93,10 +84,12 @@ def sdk_paths
 
             # Use unversioned SDK path on Big Sur to avoid issues such as:
             # https://github.com/Homebrew/homebrew-core/issues/67075
-            sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
-            if OS::Mac.version >= :big_sur && File.directory?(sdk_path)
-              sdk_settings = File.join(sdk_path, "SDKSettings.json")
-              version = JSON.parse(File.read(sdk_settings))["Version"] if File.exist?(sdk_settings)
+            # This creates an entry in `paths` whose key is the OS major version
+            sdk_path = Pathname.new("#{sdk_prefix}/MacOSX.sdk")
+            sdk_settings = sdk_path/"SDKSettings.json"
+            if sdk_settings.exist?
+              version_string = JSON.parse(sdk_settings.read)["Version"]
+              version = version_string[/(\d+)\./, 1] if version_string.present?
               paths[OS::Mac::Version.new(version)] = sdk_path if version.present?
             end
```

Comment:
```suggestion
            sdk_settings = sdk_path/"SDKSettings.json"
            if sdk_settings.exist? && 
               (sdk_settings_string = sdk_settings.read.presence) &&
               (sdk_settings_json = JSON.parse(sdk_settings_string).presence) &&
               (version_string = sdk_settings_json.fetch("Version", nil).presence) &&
               (version = version_string[/(\d+)\./, 1].presence)
              paths[OS::Mac::Version.new(version)] = sdk_path
            end
```

Maybe slightly overkill but makes it easier to follow what's going on (and nail down errors to a specific line if they occur)

Reply:
Hm, I actually wrote this so that errors would pass through silently, as in the previous way of populating `paths`, and leave the error handling to `rescue NoSDKError`.

This all happens silently, however. I suppose it would be good to catch errors in the act. Updated commits up shortly.